common/rfb/SSecurity.cxx

```diff
@@ -25,7 +25,7 @@
 using namespace rfb;
 
 SSecurity::SSecurity(SConnection* sc)
-  : sc(sc), accessRights(AccessDefault)
+  : sc(sc), accessRights(sc->getAccessRights())
```

Comment:
Won't this get overridden by some security types?

Isn't it better to let the `SSecurity` derived classes keep their own `accessRights`, and then merge it with the one in `SConnection` after authentication?

Reply:
Do you mean, by the code in `SConnection::processSecurityMsg()`?
```
   SConnection::processSecurityMsg()
```
I will need to review the code and run tests to confirm that it it would work with all security types.

Reply:
Indeed. It would probably be replaced with:

```c++
  setAccessRights(accessRights & ssecurity->getAccessRights());
```

That should merge the most restrictive set between the connection and security type rights.